Reactions/rk64/reactor_rk64.cpp

```diff
@@ -0,0 +1,296 @@
+
+  int* nstepsvec;
+  int Ncells = box.numPts();
+  const auto len = amrex::length(box);
+  const auto lo = amrex::lbound(box);
+  nstepsvec = new int[Ncells]();
+
+  amrex::ParallelFor(box, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+    amrex::Real soln_reg[NUM_SPECIES + 1];
+    amrex::Real carryover_reg[NUM_SPECIES + 1];
+    amrex::Real error_reg[NUM_SPECIES + 1];
+    amrex::Real rhs[NUM_SPECIES + 1];
+
+    amrex::Real rYsrc[NUM_SPECIES];
+    amrex::Real mass_frac[NUM_SPECIES];
+    int neq = (NUM_SPECIES + 1);
+
+#include "rkparams.H"
+
+    amrex::Real dt_rk, dt_rk_min, dt_rk_max, change_factor;
+    amrex::Real rho;
+
+    for (int sp = 0; sp < NUM_SPECIES; sp++) {
+      soln_reg[sp] = rY_in(i, j, k, sp);
+      carryover_reg[sp] = soln_reg[sp];
+      rho += rY_in(i, j, k, sp);
+      error_reg[sp] = 0.0;
+      rhs[sp] = 0.0;
+    }
+    amrex::Real rho_inv = 1.0 / rho;
+    for (int sp = 0; sp < NUM_SPECIES; sp++) {
+      mass_frac[sp] = rY_in(i, j, k, sp) * rho_inv;
+    }
+    rhs[NUM_SPECIES] = 0.0;
+    error_reg[NUM_SPECIES] = 0.0;
+    amrex::Real temp = T_in(i, j, k, 0);
+
+    amrex::Real Enrg_loc = rEner_in(i, j, k, 0) / rho;
+    auto eos = pele::physics::PhysicsType::eos();
+    if (captured_reactor_type == 1) {
+      eos.EY2T(Enrg_loc, mass_frac, temp);
+    } else {
+      eos.HY2T(Enrg_loc, mass_frac, temp);
+    }
+    soln_reg[NUM_SPECIES] = temp;
+    carryover_reg[NUM_SPECIES] = soln_reg[NUM_SPECIES];
+
+    dt_rk = dt_react / amrex::Real(captured_nsubsteps_guess);
+    dt_rk_min = dt_react / amrex::Real(captured_nsubsteps_max);
+    dt_rk_max = dt_react / amrex::Real(captured_nsubsteps_min);
+
+    amrex::Real current_time = time_init;
+    amrex::Real rhoe_init[] = {rEner_in(i, j, k, 0)};
+    amrex::Real rhoesrc_ext[] = {rEner_src_in(i, j, k, 0)};
+
+    for (int sp = 0; sp < NUM_SPECIES; sp++) {
+      rYsrc[sp] = rY_src_in(i, j, k, sp);
+    }
+
+    int nsteps = 0;
+    while (current_time < time_out) {
+      for (int sp = 0; sp < neq; sp++) {
+        error_reg[sp] = 0.0;
+      }
+      for (int stage = 0; stage < nstages_rk64; stage++) {
+        fKernelSpec(
+          0, 1,current_time - time_init, captured_reactor_type, soln_reg, rhs,
+          rhoe_init, rhoesrc_ext, rYsrc);
+
+        for (int i = 0; i < neq; i++) {
+          error_reg[i] += err_rk64[stage] * dt_rk * rhs[i];
+          soln_reg[i] = carryover_reg[i] + alpha_rk64[stage] * dt_rk * rhs[i];
+          carryover_reg[i] = soln_reg[i] + beta_rk64[stage] * dt_rk * rhs[i];
+        }
+      }
+
+      current_time += dt_rk;
+      nsteps++;
+
+      amrex::Real max_err = tinyval;
+      for (int i = 0; i < neq; i++) {
+        if (fabs(error_reg[i]) > max_err) {
+          max_err = fabs(error_reg[i]);
+        }
+      }
+
+      if (max_err < captured_abstol) {
+        change_factor =
+          betaerr_rk64 * pow((captured_abstol / max_err), exp1_rk64);
+        dt_rk = std::min(dt_rk_max, dt_rk * change_factor);
```

Comment:
Let's use `amrex::min<amrex::Real>` and `max`.

Reply:
done